File: backend/src/main/java/az/project/VocabularyBuilder/vocabulary/dto/QuizViewDto.java
package az.project.VocabularyBuilder.vocabulary.dto;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

public record QuizViewDto(
        // --- Ogólne Statystyki Quizu ---
        UUID quizUuid,

        int scoreCorrect,
        int scoreTotal,

        // Dokładność procentowa (np. 85.71)
        double accuracyPercent,

        int durationSeconds,
        LocalDateTime dateCompleted,

        // --- Szczegółowe Wyniki Pytań ---
        List<QuizItemView> itemResults
) {
}

File: frontend/src/app/features/vocabulary-quiz/models/quizViewDto.ts
import { MasteryStatus } from "../../../shared/enums/masterStatus";
import { PartOfSpeech } from "../../../shared/enums/partOfSpeech";

export interface QuizViewDto {
  quizUuid: string;
  scoreCorrect: number; 
  scoreTotal: number;
  accuracyPercent: number;
  durationSeconds: number;
  dateCompleted: string;
  itemResults: QuizItemView[]; 
}

export interface QuizItemView {
 
entryId: number;
  wordPhraseEn: string;
  partOfSpeech: PartOfSpeech;
  isCorrect: boolean;
  userAnswer: string | null;
  correctAnswer: string;
  finalMasteryStatus: MasteryStatus;
}
